consensus/src/experimental/buffer_item.rs

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::state_replication::StateComputerCommitCallBackType;
+use consensus_types::{common::Author, executed_block::ExecutedBlock};
+use diem_crypto::ed25519::Ed25519Signature;
+use diem_types::{
+    account_address::AccountAddress,
+    block_info::BlockInfo,
+    ledger_info::{LedgerInfo, LedgerInfoWithSignatures},
+    validator_verifier::ValidatorVerifier,
+};
+use itertools::zip;
+use std::collections::BTreeMap;
+
+// we differentiate buffer items at different stages
+// for better code readability
+pub struct OrderedBufferItem {
+    pub cache: BTreeMap<AccountAddress, Ed25519Signature>,
```

Comment:
maybe pending_votes is more explicit than cache

Reply:
resolved

Reply:
resolved